models/image.py

```diff
@@ -813,6 +883,54 @@ def from_new_and_ref(cls, new_image, ref_image):
         # Note that "data" is not filled by this method, also the provenance is empty!
         return output
 
+    def align_images(self):
+        """Align the upstream_images to one of the images pointed to by image_index.
+
+        The parameters of the alignment must be given in the parameters attribute
+        of this Image's Provenance.
+
+        The index to which the images are aligned is given by the parameters, using
+        the "to_index" key, which can be "first" or "last".
+
+        The resulting images are saved in _aligned_images, which are not saved
+        to the database. Note that each aligned image is also referred to by
+        a global variable under the ImageAligner.temp_images list.
+        """
+        from pipeline.alignment import ImageAligner  # avoid circular import
+        if self.provenance is None or self.provenance.parameters is None:
+            raise RuntimeError('Cannot align images without a Provenace with legal parameters!')
+        if 'alignment' not in self.provenance.parameters:
+            raise RuntimeError('Cannot align images without an "alignment" dictionary in the Provenance parameters!')
+
+        if self.provenance.parameters['alignment']['to_index'] == 'first':
+            image_index = 0
+        elif self.provenance.parameters['alignment']['to_index'] == 'last':
+            image_index = -1
+
+        if self._aligner is None:
+            self._aligner = ImageAligner(**self.provenance.parameters['alignment'])
+        else:
+            self._aligner.pars.override(**self.provenance.parameters['alignment'])
+
+        # verify all products are loaded
+        for im in self.upstream_images:
+            if im.sources is None or im.wcs is None or im.zp is None:
```

Comment:
Does alignment actually need the zp?  I believe it only really uses sources.  (It may not even use the wcs.)